saltproc/depcode.py

```diff
@@ -610,21 +624,40 @@ def sss_meta_zzz(self, nuc_code):
             zzaaam = nuc_code
         return int(zzaaam)
 
+    def switch_to_next_geometry(self):
+        """Inserts line with path to next Serpent geometry file at the
+        beginning of the Serpent iteration input file.
+        """
+        geo_line_n = 5
+        f = open(self.iter_inputfile, 'r')
+        data = f.readlines()
+        f.close()
+
+        current_geo_file = data[geo_line_n].split('\"')[1]
+        current_geo_idx = self.geo_files.index(current_geo_file)
+        try:
+            new_geo_file = self.geo_files[current_geo_idx + 1]
+        except IndexError:
+            print('No more geometry files available \
```

Comment:
`switch_to_next_geometry` is called based on the following flow control in `app.py`:
```
if simulation.adjust_geo and simulation.read_k_eds_delta(dep_step):
             depcode.switch_to_next_geometry()

```
`read_k_eds_delta` returns `True` if it is predicted that the reactor will go sub critical at the next serpent run. So we would change the reactor geometry (i.e. control rod depth) to ensure that the simulation remains critical. So if there is no geometry file to switch to, then the reactor will be subcritical hence the error message.

I do agree that the location of the error message in the source code is probably not appropriate. I'll see if it can be moved to the flow control in `app.py`.

This feature of SaltProc should also be documented.

Reply:
@gwenchee i've created an issue regarding the exceptions (#118). I've also added a section to issue #109 about documenting the geometry switching feature.

Reply:
sounds good